internal/controllers/clusterextension_controller.go

```diff
@@ -550,21 +552,54 @@ func (r *ClusterExtensionReconciler) generateBundleDeploymentForUnpack(bundlePat
 				Image: &rukpakv1alpha2.ImageSource{
 					Ref:                   bundlePath,
 					InsecureSkipTLSVerify: isInsecureSkipTLSVerifySet(ce),
+					CertificateData:       certData,
 				},
 			},
 		},
 	}
 }
 
 func isInsecureSkipTLSVerifySet(ce *ocv1alpha1.ClusterExtension) bool {
-	if ce == nil {
+	if ce == nil || ce.Spec.RegistryTLS == nil {
 		return false
 	}
-	value, ok := ce.Annotations[bundleConnectionAnnotation]
-	if !ok {
-		return false
+	return ce.Spec.RegistryTLS.InsecureSkipTLSVerify
+}
+
+func (r *ClusterExtensionReconciler) getCertificateData(ctx context.Context, ce *ocv1alpha1.ClusterExtension) (string, error) {
+	if ce == nil || ce.Spec.RegistryTLS == nil || ce.Spec.RegistryTLS.CertificateSecretRef == nil {
+		return "", nil
+	}
+	secretName := getNamespacedName(*ce.Spec.RegistryTLS.CertificateSecretRef)
+	var secret = &corev1.Secret{}
+	if err := r.Reader.Get(ctx, secretName, secret); err != nil {
+		return "", fmt.Errorf("unable to get secret %v: %w", secretName, err)
+	}
```

Comment:
It looks like `r.Reader` is a live client (no cache). How often are we going to be hitting API server with this? Every reconcile?

Reply:
Correct. We don't want to cache _all_ the Secrets of the system.

IT should be accessed only when we actually attempt to install.